File: Longest Sub-Array with Sum K.cpp
class Solution{
    public:
    int lenOfLongSubarr(int A[],  int N, int K) 
    { 
        // Complete the function
        map<int,int> mp;
        mp.insert({0,-1});
      
        int ans = 0, sum = 0;
        
        for(int i=0; i<N; i++)
        {
            sum += A[i];
            
            if(mp.find(sum-K) != mp.end())
            ans = max(ans, (i-mp[sum-K]));
            
            if(mp.find(sum) == mp.end())
            mp[sum] = i;
        }
        
        return ans;
    } 
  
};

File: Longest Sub-Array with Sum K.java
class Solution{
    
   
    // Function for finding maximum and value pair
    public static int lenOfLongSubarr (int A[], int N, int K) {
        //Complete the function
        Map<Integer,Integer> map=new HashMap<>();
        int sum=0;
        int length=0;
        for(int i=0;i<N;i++){
            sum+=A[i];
            if(sum==K){
                length=i+1;
            }else if(map.containsKey(sum-K)){
                length=Math.max(length,i-map.get(sum-K));
            }
            
            if(!map.containsKey(sum)){
                map.put(sum,i);
            }
        }
        return length;
    }
    
    
}
